lib/cli/provider/index.js

```diff
@@ -154,6 +159,9 @@ module.exports.handler = function(argv) {
   , filter: function(device) {
       return argv.serial.length === 0 || argv.serial.indexOf(device.id) !== -1
     }
+    , exclude: function(device) {
```

Comment:
Agreed, we can handle both inclusion and exclusion in filter function. 

For second problem. I just did an experiment and found a workaround. I have a device with serial name KNSCMJJFZ9RCNFHE, the device can be excluded with pattern "^KNSCMJJFZ9RCNFHE$", but not with the pattern "^KNSCMJJFZ9RCNFHEX$" , so I think the options of "exclude/include-pattern" might be redundant thought they are more explicit to users who may not know Regexp.